Src/Coravel/Scheduling/Schedule/Event/ScheduledEvent.cs

```diff
@@ -18,6 +18,7 @@ public class ScheduledEvent : IScheduleInterval, IScheduledEventConfiguration
         private bool _preventOverlapping = false;
         private string _eventUniqueID = null;
         private IServiceScopeFactory _scopeFactory;
+        private Func<Task<bool>> predicate;
```

Comment:
I wonder if naming this something else would give it better meaning - `_whenPredicate` ?